doc/ir-semantics.text

```diff
@@ -111,18 +111,15 @@ Update (partial) function, returns an H:
 
 ## Osr transformation of environment
 
-  materialize P H E (mut x = e) = (x ↦ (a ↦ v))
-    where a - fresh
-      and v := eval P H E e
-  materialize P H E _           = ()
-
-  osr-env P H E M (const x = e) = (x ↦ val v)
+  create-env H E (const x = e) = (∅, [x ↦ val v])
     where v := eval P H E e
-  osr-env P H E M (mut x)       = (x ↦ ⊥)
-  osr-env P H E M (mut x = &y)  = (x ↦ ref a)
+  create-env H E (mut x)       = ([a ↦ ⊥], [x ↦ ref a])
+    where a fresh
```

Comment:
Oh, you are right. That was a lingering bug. I agree with your point. Lets go this way.